### PR TITLE
Fix/sign up

### PR DESCRIPTION
**Pull request by JeremyDec**
| Question         | Answer
| ---------------- | ---
| Branch?          | develop
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | no
| Tickets          | Fix #MANAGER-4598 #MANAGER-4461 #MANAGER-4462
| License          | BSD 3-Clause

## Description

Fixes for Sign up account form

### CODE REVIEW

File: packages/manager/apps/sign-up/src/index.html [hunk 1]
```diff
@@ -21,12 +21,12 @@
                     data-ng-href="{{ $ctrl.logoUrl }}"
                 >
                     <img
-                        src="assets/img/logo-ovh-baseline.png"
+                        src="assets/img/Logo OVHcloudTM.png"
```
Comment (antleblanc) [inline packages/manager/apps/sign-up/src/index.html:24]: This logo should be displayed for a single region.

Could you please keep the file naming convention? (no space, lower case and `-`).

File: packages/manager/apps/sign-up/src/index.html [hunk 2]
```diff
@@ -21,12 +21,12 @@
...
                         srcset="
-                            assets/img/logo-ovh-baseline@2x.png 2x,
-                            assets/img/logo-ovh-baseline@3x.png 3x
+                            assets/img/Logo OVHcloudTMx2.png 2x,
+                            assets/img/Logo OVHcloudTMx3.png 3x
                         "
-                        alt="OVHcloud - Innovation for freedom"
+                        alt="OVHcloud™"
```
Comment (antleblanc) [inline packages/manager/apps/sign-up/src/index.html:29]: It must be specific for a given region.

File: packages/manager/apps/sign-up/src/index.html (hunk 1 above)
Comment (antleblanc) [reply packages/manager/apps/sign-up/src/index.html:24]: Extra request: Do you think that the `<title>` of the page could be also updated from `OVH` to `OVHcloud`?

Comment (JeremyDec) [reply packages/manager/apps/sign-up/src/index.html]: @antleblanc yes, done :)

**Comment by JeremyDec** [comment]
@FredericEspiau / @frenautvh , logo files has been replaced and webpack config has changed to load correct assets based on region

**Comment by ovh-cds** [comment]
CDS Report build#12236.0 ✘  
* 
  * Build ✘

**Comment by ovh-cds** [comment]
CDS Report build#12236.1 ✘  
* 
  * Build ✘

**Comment by ovh-cds** [comment]
CDS Report build#12236.2 ✘  
* 
  * Build ✘

**Comment by ovh-cds** [comment]
CDS Report build#12237.0 ✘  
* 
  * Build ✘

**Comment by ovh-cds** [comment]
CDS Report build#12237.1 ✘  
* 
  * Build ✘

**Comment by ovh-cds** [comment]
CDS Report build#12237.2 ✘  
* 
  * Build ✘

**Comment by ovh-cds** [comment]
CDS Report build#12238.0 ✘  
* 
  * Build ✘

**Comment by ovh-cds** [comment]
CDS Report installation#12259.0 ■   
* Stage 1
  * Install ■

**Comment by ovh-cds** [comment]
CDS Report installation#12259.0 ■   
* Stage 1
  * Install ■

File: packages/manager/modules/sign-up/src/form/details/details.controller.js [hunk 3]
```diff
@@ -151,14 +184,7 @@ export default class SignUpDetailsCtrl {
...
       // So... be sure that the phone validation is done.
-      const phoneModel = this.formCtrl.phone;
-      const modelValue = phoneModel.$modelValue;
-      const viewValue = phoneModel.$viewValue;
-      const phoneModelValid = phoneModel.$validators.pattern(
-        modelValue,
-        viewValue,
-      );
-      phoneModel.$setValidity('pattern', phoneModelValid);
+      this.constructor.setInputValidity('phone', this.formCtrl);
```
Comment (FredericEspiau) [inline packages/manager/modules/sign-up/src/form/details/details.controller.js:187]: Is the comment just above still valid ?

File: packages/manager/apps/sign-up/webpack.config.js [hunk 4]
```diff
@@ -19,7 +19,15 @@ module.exports = (env = {}) => {
             ),
             to: 'flag-icon-css/flags/4x3',
           },
-          { from: path.resolve(__dirname, 'src/assets/img'), to: 'assets/img' },
+          {
+            from: path.resolve(
+              __dirname,
+              `src/assets/img/logo${
+                process.env.REGION === 'US' ? '/trademark' : '/classic'
```
Comment (JeremyDec) [inline packages/manager/apps/sign-up/webpack.config.js:26]: Still working on it, build in US region doesn't take care of this

File: packages/manager/modules/sign-up/src/form/details/details.controller.js (hunk 3 above)
Comment (JeremyDec) [reply packages/manager/modules/sign-up/src/form/details/details.controller.js:187]: Yes it is still valid, as it still sets validity of phone based on the pattern

**Comment by ovh-cds** [comment]
CDS Report installation#12286.0 ■   
* Stage 1
  * Install ■

**Comment by ovh-cds** [comment]
CDS Report installation#12286.0 ■   
* Stage 1
  * Install ■

**Comment by JeremyDec** [comment]
@FredericEspiau => final change into build files. Nothing more since your approval.
@antleblanc => see the two last commits for your remarks.

File: packages/manager/modules/sign-up/src/form/details/details.controller.js [hunk 5]
```diff
@@ -115,6 +115,35 @@ export default class SignUpDetailsCtrl {
...
+    const inputModelValid = inputField.$validators.pattern(
+      inputModelValue,
+      inputViewValue,
+    );
+
+    inputField.$setValidity('pattern', inputModelValid);
+  }
+
+  preselectLanguage() {
+    if (get(this.signUpFormCtrl, 'rules.language.in.length', 0) === 1) {
```
Comment (FredericEspiau) [inline packages/manager/modules/sign-up/src/form/details/details.controller.js:139]: nit:

```suggestion
this.signUpFormCtrl.model.language = get(this.signUpFormCtrl, 'rules.language.in.length', 0) === 1 && this.signUpFormCtrl.rules.language.in[0].value || null
    if () {
```

File: packages/manager/modules/sign-up/src/form/details/details.controller.js [hunk 6]
```diff
@@ -115,6 +115,35 @@ export default class SignUpDetailsCtrl {
...
+      const [uniqueLanguage] = this.signUpFormCtrl.rules.language.in;
```
Comment (Jisay) [inline packages/manager/modules/sign-up/src/form/details/details.controller.js:140]: Do you know why it's not autoselected by this part of code?
https://github.com/ovh/manager/blob/8ad05e5e3a1a68e579ad64ae1f02198d8b78cc4f/packages/manager/modules/sign-up/src/form/form.controller.js#L95-L100

Comment (JeremyDec) [reply packages/manager/modules/sign-up/src/form/details/details.controller.js:140]: It seems that the language option list contains a blank item. Don't know if it's related to this, but I will check again

**Comment by JeremyDec** [comment]
Since previous approval => 7f3b1cb  has been updated to trigger on area change.
Additional refactor commit => 338d507

**Pull request merged**